portfolio/src/main/webapp/script.js

```diff
@@ -1,6 +1,23 @@
 // fetches comments from the server and populates the html element with id 'comments'
 async function getComments() {
     const response = await fetch('/comments');
-    const json = await response.text(); // collection of comments
-    document.getElementById('comments').innerText = json;
+    const comments = await response.json(); // collection of comments
+    const container = document.getElementById('comments');
+    insertComments(comments, container);
+}
+
+// Accepts an array of { content: String, timestamp: long }, and adds them to the 'comments' element

```

Comment:
Nit: personal preference, I would add full stop to all method description since they would be read by others (inline comments are more casual).